crates/sui-indexer-alt-restorer/src/snapshot.rs

```diff
@@ -0,0 +1,188 @@
+            ),
+        ));
+
+        futures::stream::iter(input_files)
+            .try_for_each_spawned(
+                self.restore_args.concurrency,
+                |(bucket, (part_num, file_metadata))| {
+                    let epoch_dir = epoch_dir.clone();
+                    let remote_object_store = remote_object_store.clone();
+                    let sha3_digests = sha3_digests.clone();
+                    let bar = move_object_progress_bar.clone();
+                    let db = self.db.clone();
+                    let next_cp = self.next_checkpoint_after_epoch;
+
+                    async move {
+                        let mut conn = db.connect().await?;
+                        let object_file_path = file_metadata.file_path(&epoch_dir);
+                        let (bytes, _) = download_bytes(
+                            remote_object_store,
+                            &file_metadata,
+                            epoch_dir,
+                            sha3_digests,
+                            &&bucket,
+                            &part_num,
+                            Some(512), // max_timeout_secs
+                        )
+                        .await;
+                        info!(
+                            path = ?object_file_path,
+                            "Finished downloading move object file"
+                        );
+                        let mut object_infos = vec![];
+                        LiveObjectIter::new(&file_metadata, bytes.clone()).map(|obj_iter| {
+                            for object in obj_iter {
+                                match object {
+                                    LiveObject::Normal(obj) => {
+                                        let object_info =
+                                            StoredObjInfo::from_object(&obj, next_cp as i64);
+                                        object_infos.push(object_info);
+                                    }
+                                    LiveObject::Wrapped(_) => {}
+                                }
+                            }
+                        })?;
+                        let object_infos =
+                            object_infos.into_iter().collect::<Result<Vec<_>, _>>()?;
+
+                        // NOTE: chunk to avoid hitting the PG limit
+                        let object_info_count = object_infos.len();
+                        let chunk_size: usize = i16::MAX as usize / StoredObjInfo::FIELD_COUNT;
+                        for chunk in object_infos.chunks(chunk_size) {
+                            diesel::insert_into(obj_info::table)
+                                .values(chunk)
+                                .on_conflict_do_nothing()
+                                .execute(&mut conn)
+                                .await?;
+                        }
+                        bar.inc(1);
+                        info!(count = object_info_count, "Restored move objects");
```

Comment:
nit: is it going to mess with the progress bar to also print out this `info!` statement?

Reply:
the bar updates will show on multiple lines like 
```
[00:03:17] ███████████░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ 70 out of 700 ...
INFO ...
[00:03:19] ███████████░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ 71 out of 700...
```
it's indeed neater to have bar updated in place, replaced info! with bar.set_message for that.